### PR TITLE
PM-11150: Add import logins step 1

### PR DESCRIPTION
**Pull request by matt-livefront**
## 🎟️ Tracking

[PM-11150](https://bitwarden.atlassian.net/browse/PM-11150)

## 📔 Objective

Adds first step of import logins instructions. This also switches to using a new feature flag separately from the create account flow so they can be enabled separately if needed.

## 📸 Screenshots

Screenshot 2024-10-10 at 2 14 45 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11150]: https://bitwarden.atlassian.net/browse/PM-11150?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Platform/Application/Views/NumberedList.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+struct NumberedList<Content: View>: View {
+    // MARK: Properties
+
+    /// The content to display in the numbered list. Each child view will receive it's own number.
+    let content: Content
+
+    // MARK: View
+
+    var body: some View {
+        _VariadicView.Tree(Layout()) {
```
Comment (phil-livefront) [inline BitwardenShared/UI/Platform/Application/Views/NumberedList.swift:19]: is this part of swiftui's private view api?  I should make a ticket to use this within `PasswordAutoFillView`

**Review by phil-livefront — APPROVED** [review]

File: BitwardenShared/UI/Platform/Application/Views/NumberedList.swift (hunk 1 above)
Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Application/Views/NumberedList.swift:19]: Yeah, it is a private SwiftUI API, but it's used pretty extensively within SwiftUI. There's a few places in the app that we could make use of this after it's in.

File: BitwardenShared/UI/Vault/Vault/ImportLogins/ImportLoginsView.swift [hunk 2]
```diff
@@ -35,20 +56,83 @@ struct ImportLoginsView: View {
...
+    /// - Parameters:
+    ///   - step: The step number of this page.
+    ///   - totalSteps: The total number of steps.
+    ///   - title: The title of the step.
+    ///   - list: A closure that returns the views to display in a numbered list.
+    ///
+    @ViewBuilder
+    private func stepView(
+        step: Int,
+        totalSteps: Int,
```
Comment (fedemkr) [inline BitwardenShared/UI/Vault/Vault/ImportLogins/ImportLoginsView.swift:85]: 🤔 Could this `totalSteps` be extracted to a constant at the file level? Maybe I'm misunderstanding this but I feel this value will be the same every time this func is called.

File: BitwardenShared/UI/Platform/Application/Views/NumberedList.swift
Comment (fedemkr) [inline BitwardenShared/UI/Platform/Application/Views/NumberedList.swift:1]: 🤔 Shall a comment or something be added to call out this uses a non exactly public API from Apple?

File: BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -1016,3 +1016,10 @@
 "YouCanReturnToCompleteThisStepAnytimeInVaultUnderSettings" = "You can return to complete this step anytime in Vault under Settings.";
 "DoYouHaveAComputerAvailable" = "Do you have a computer available?";
 "DoYouHaveAComputerAvailableDescriptionLong" = "The following instructions will guide you through importing logins from your desktop or laptop computer.";
+"StepXOfY" = "Step %1$@ of %2$@";
+"ExportYourSavedLogins" = "Export your saved logins";
+"ExportLoginsStep1" = "On your computer, **log in to your current browser or password manager.**";
+"ExportLoginsStep2" = "**Export your passwords.** This option is usually found in your settings.";
+"ExportLoginsStep3" = "**Select Import data** in the web app, then Done to finish syncing.";
+"ExportLoginsStep3Subtitle" = "You’ll delete this file after import is complete.";
```
[the remark this answers is not in the document]
Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings:1024]: Sure, I made those changes here: https://github.com/bitwarden/ios/pull/1044

File: BitwardenShared/UI/Vault/Vault/ImportLogins/ImportLoginsView.swift (hunk 2 above)
Comment (matt-livefront) [reply BitwardenShared/UI/Vault/Vault/ImportLogins/ImportLoginsView.swift:85]: Good point, changes are in latest PR.

File: BitwardenShared/UI/Platform/Application/Views/NumberedList.swift
Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Application/Views/NumberedList.swift:1]: I added a comment in my latest PR.

**Pull request merged**